Resfeber/CoreData/CoreDataStack.swift

```diff
@@ -10,63 +10,64 @@ import Foundation
-    let container = NSPersistentContainer(name: CoreDataStack.modelName, managedObjectModel: CoreDataStack.model)
-    container.loadPersistentStores { _, error in
-      if let error = error as NSError? {
-        fatalError("Unresolved error \(error), \(error.userInfo)")
-      }
+    public static let model: NSManagedObjectModel = {
+        let modelURL = Bundle.main.url(forResource: modelName, withExtension: "momd")!
+        return NSManagedObjectModel(contentsOf: modelURL)!
+    }()
+    
+    private init() {
     }
-    return container
-  }()
-
-  public func newDerivedContext() -> NSManagedObjectContext {
-    let context = storeContainer.newBackgroundContext()
-    return context
-  }
-
-  public func saveContext() {
-    saveContext(mainContext)
-  }
-
-  public func saveContext(_ context: NSManagedObjectContext) {
-    if context != mainContext {
-      saveDerivedContext(context)
-      return
+    
+    public lazy var mainContext: NSManagedObjectContext = {
+        return storeContainer.viewContext
+    }()
+    
+    public lazy var storeContainer: NSPersistentContainer = {
+        let container = NSPersistentContainer(name: CoreDataStack.modelName, managedObjectModel: CoreDataStack.model)
+        container.loadPersistentStores { _, error in
+            if let error = error as NSError? {
+                fatalError("Unresolved error \(error), \(error.userInfo)")
+            }
+        }
+        return container
+    }()
+    
+    public func newDerivedContext() -> NSManagedObjectContext {
+        let context = storeContainer.newBackgroundContext()
+        return context
     }
-
-    context.perform {
-      do {
-        try context.save()
-      } catch let error as NSError {
-        fatalError("Unresolved error \(error), \(error.userInfo)")
-      }
+    
+    public func saveContext() {
+        saveContext(mainContext)
     }
-  }
-
-  public func saveDerivedContext(_ context: NSManagedObjectContext) {
-    context.perform {
-      do {
-        try context.save()
-      } catch let error as NSError {
-        fatalError("Unresolved error \(error), \(error.userInfo)")
-      }
-
-      self.saveContext(self.mainContext)
+    
+    public func saveContext(_ context: NSManagedObjectContext) {
+        if context != mainContext {
+            saveDerivedContext(context)
+            return
+        }
+        
+        context.perform {
+            do {
+                try context.save()
+            } catch let error as NSError {
+                fatalError("Unresolved error \(error), \(error.userInfo)")
+            }
+        }
+    }
+    
+    public func saveDerivedContext(_ context: NSManagedObjectContext) {
+        context.perform {
+            do {
+                try context.save()
+            } catch let error as NSError {
+                fatalError("Unresolved error \(error), \(error.userInfo)")
+            }
+            
+            self.saveContext(self.mainContext)
+        }
```

Comment:
I'm actually not sure why you need all three of these. What I would recommend is that you just have one save function that has the context parameter with a default value of `CoreDataStack.mainContext` that way if you call it with no parameters, it will just save on the main context, and if you do call it with a parameter, it will call save on that context